### DISCUSSION TITLE
nextjs-Canary Keeps throwing Undefined Error !

### DISCUSSION THREAD

**Discussion by Alii-isk** [vercel/next.js#28338] [Help]
### What version of Next.js are you using?

11.1.1-canary.13

### What version of Node.js are you using?

v14.15.4

### What browser are you using?

brave

### What operating system are you using?

windows 10

### How are you deploying your application?

still on development server

### Describe the Bug

i am having this issue on  `dbConnect.js ` it keeps throwing the error although there is none but when i change back to `"next": "11.0.1"` it start working just fine.
i think it something related to .env
```

import mongoose from 'mongoose'

const MONGODB_URI = process.env.MONGO_URI
console.log(MONGODB_URI ) //this one is not undefined
if (!MONGODB_URI) {
    throw new Error( 
        'Please define the MONGODB_URI environment variable inside .env.local'
    )
...

```}

### Expected Behavior

there should not be any error

### To Reproduce

```
    import mongoose from 'mongoose'
    
    const MONGODB_URI = process.env.MONGO_URI
    if (!MONGODB_URI) {
        throw new Error(
            'Please define the MONGODB_URI environment variable inside .env.local'
        )
    }
    
  
    let cached = global.mongoose
    
    if (!cached) {
        cached = global.mongoose = { conn: null, promise: null }
    }
    
    async function dbConnect() {
        if (cached.conn) {
            return cached.conn
        }
    
        if (!cached.promise) {
            const opts = {
                useNewUrlParser: true,
                useUnifiedTopology: true,
                bufferCommands: false,
                bufferMaxEntries: 0,
                useFindAndModify: false,
                useCreateIndex: true,
            }
    
            cached.promise = mongoose.connect(MONGODB_URI, opts).then((mongoose) => {
                console.log(":::: New Connection to db!")
                return mongoose
            })
        }
        cached.conn = await cached.promise
        return cached.conn
    }
    
    export function jsonify(obj) {
        return JSON.parse(JSON.stringify(obj));
    }
    
    export default dbConnect

```

**Comment by dikwickley** [comment]
Your .env file is probably not being read.
You can use try
1) Making sure the naming is correct (of files and variables)
2) Removing .next and node_modules and lock.json and trying to install all packages again.
3) use the dotenv package (https://www.npmjs.com/package/dotenv)

**Comment by Regisnut** [comment]
Did you solve your issue @[Alii-isk](https://github.com/Alii-isk)
got the same issue 
Error: Please define the MONGODB_URI environment variable inside .env.local

What version of Next.js are you using?
 "next": "^12.0.10",

What version of Node.js are you using?
v14.15.1

What browser are you using?
chrome

What operating system are you using?
macOs Bigsur

How are you deploying your application?
still on development server / vercel on production

**Reply by Alii-isk** [reply]
I ended up changing nextjs version my project is currently using 12.0.7 version and i have no issues

```
"dependencies": {
     ...
    "next": "12.0.7",
    ...
 
    
  }

```
